### PR TITLE
Added support for setting from_name

### PR DESCRIPTION
**Pull request by gonguinguen**
Accordingly to official documentation [1], "from" parameter can be just a string (an email address), or a JSON dictionary, with the keys "name" and "email". So I've extended the send method to check for the "from_name" parameter, and if it's present, sending a dictionary instead of just the email.

[1] https://developers.sparkpost.com/api/#/reference/transmissions

### CODE REVIEW

File: sparkpost/transmissions.py [hunk 1]
```diff
@@ -39,7 +39,13 @@ def _translate_keys(self, **kwargs):
             kwargs.get('use_draft_template', False)
         model['content']['reply_to'] = kwargs.get('reply_to')
         model['content']['subject'] = kwargs.get('subject')
-        model['content']['from'] = kwargs.get('from_email')
+        if kwargs.get('from_name'):
```
Comment (rajumsys) [inline sparkpost/transmissions.py]: Thank you very much for the PR. 

You seem have some trailing whitespaces which causing the build failure due to linting issue. Just removing the trailing whitespaces should make the build green, i hope :).

File: sparkpost/transmissions.py [hunk 2]
```diff
@@ -39,7 +39,13 @@ def _translate_keys(self, **kwargs):
...
+            model['content']['from'] = {
+                'name': kwargs.get('from_name'),
```
Comment (rajumsys) [inline sparkpost/transmissions.py]: Just curious what will it pass if `get` returns `None` (it's default value). Will it pass through that to API as `none` or keeps it empty?

File: sparkpost/transmissions.py (hunk 1 above)
Comment (gonguinguen) [inline sparkpost/transmissions.py]: Sorry about that, I've just removed spaces and did a simple fix to send an empty string if from_name isn't specifyied.
I did the commit. I don't know if I should do another pull request. It's the first time I use github as a contributor :P

File: sparkpost/transmissions.py [hunk 3]
```diff
@@ -39,7 +39,13 @@ def _translate_keys(self, **kwargs):
             kwargs.get('use_draft_template', False)
         model['content']['reply_to'] = kwargs.get('reply_to')
         model['content']['subject'] = kwargs.get('subject')
-        model['content']['from'] = kwargs.get('from_email')
+        if kwargs.get('from_name'):
+            model['content']['from'] = {
+                'name': kwargs.get('from_name') or '',
```
Comment (rajumsys) [inline sparkpost/transmissions.py]: this isn't necessary because you've already checked it's truthy-ness in line 42. I commented that without noticing that one and I deleted after I found it's not necessary. Sorry for confusion. 

You don't need to create another PR. Just commit and push the same branch which will auto update the PR. Thanks again. 

Can you also possibly add unit tests covering you tests? `transmissions` module is now missing some coverage.

**Comment by richleland** [comment]
Thanks for the PR @gonguinguen - so this is a huge documentation fail on my part - you can supply a friendly from in the existing `from_email` field. For example the following are both valid:

```python
sp.transmissions.send(..., from_email='testing@example.com', ...)
```

```python
sp.transmissions.send(..., from_email='Test Email <testing@example.com>', ...)
```

The `_extract_recipients` method handles converting either of these into the proper format for the API. We just need to document it. I'll create another issue to handle updating the docs.

**Comment by gonguinguen** [comment]
Awesome!! It works like a charm.
Then I think the changes I commited should be discarded. If that's
something for what I'm responsble, then let me know (sorry, I'm new on
GitHub, I'm catching up)

2016-03-07 10:50 GMT-03:00 Richard Leland <notifications@github.com>:

**Comment by richleland** [comment]
I'll go ahead and close it - glad this works for you and sorry you had to waste some time!!
